src/hotspot/share/gc/shared/memAllocator.cpp

```diff
@@ -70,19 +70,21 @@ class MemAllocator::Allocation: StackObj {
 public:
   Allocation(const MemAllocator& allocator, oop* obj_ptr)
     : _allocator(allocator),
-      _thread(JavaThread::current()),
       _obj_ptr(obj_ptr),
       _overhead_limit_exceeded(false),
       _allocated_outside_tlab(false),
       _allocated_tlab_size(0),
       _tlab_end_reset_for_sample(false)
   {
+    assert(Thread::current()->is_Java_thread(), "must be used by JavaThreads only");
```

Comment:
This first assert duplicates the one done by `JavaThread::cast()` given the other assert just below.